packages/smooth_app/lib/pages/product/product_page/raw_data/product_raw_data_category_item.dart

```diff
@@ -0,0 +1,116 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+import 'package:smooth_app/generic_lib/design_constants.dart';
+import 'package:smooth_app/pages/product/product_page/raw_data/category_label_ext.dart';
+import 'package:smooth_app/pages/product/product_page/raw_data/models/product_raw_data_category.dart';
+import 'package:smooth_app/pages/product/product_page/raw_data/product_raw_data_category_elements_item.dart';
+import 'package:smooth_app/resources/app_icons.dart' as icons;
+import 'package:smooth_app/themes/smooth_theme.dart';
+import 'package:smooth_app/themes/smooth_theme_colors.dart';
+import 'package:smooth_app/themes/theme_provider.dart';
+
+class ProductRawDataCategoryItem extends StatelessWidget {
+  const ProductRawDataCategoryItem(this.category, this.onEditTap, {this.controller});
+
+  final ProductRawDataCategory category;
+  final ScrollController? controller;
+  final GestureTapCallback? onEditTap;
+
+  @override
+  Widget build(BuildContext context) {
+    final AppLocalizations appLocalizations = AppLocalizations.of(context);
+
+    return Column(
+        crossAxisAlignment: CrossAxisAlignment.start,
+        children: <Widget>[
+          _ProductRawDataCategoryTile(
+            category.category.toIcon(),
+            category.category.toL10nString(appLocalizations),
+            onEditTap,
+          ),
+          CategoryElementsListView(
+            elements: category.rawDatas,
+            controller: controller,
+          ),
+        ]);
+  }
+}
+
+class _ProductRawDataCategoryTile extends StatelessWidget {
+  const _ProductRawDataCategoryTile(this.icon, this.label, this.onEditTap);
+
+  final Widget icon;
+  final String label;
+  final GestureTapCallback? onEditTap;
+
+  @override
+  Widget build(BuildContext context) {
+    final bool lightTheme = context.lightTheme();
+    final Color contentColor = lightTheme
+        ? context.extension<SmoothColorsThemeExtension>().primaryBlack
```

Comment:
context.extension requires to navigate in the tree.
Please cache it to avoid multiple passes